crates/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -95,17 +97,31 @@ where
             .into_iter()
             .map(|call_arg| load_call_arg(state_view, &mut object_owner_map, call_arg))
             .collect::<Result<_, ExecutionError>>()?;
-        let gas = load_object(
+        let mut gas = load_object(
```

Comment:
I do not know how well this is going to work with smashing gas the way I have it, but we can figure that out later

Reply:
to put more thoughts maybe later we will move smashing here and check at the end that is all good (no reset to temporary store was performed)